server/map_rpc_server.go

```diff
@@ -366,6 +349,34 @@ func (t *TrillianMapServer) SetLeaves(ctx context.Context, req *trillian.SetMapL
 	return &trillian.SetMapLeavesResponse{MapRoot: newRoot}, nil
 }
 
+func (t *TrillianMapServer) newTXRunner(tree *trillian.Tree, tx storage.MapTreeTX) merkle.TXRunner {
+	if t.opts.UseSingleTransaction {
+		return &singleTXRunner{tx: tx}
+	}
+	return &multiTXRunner{tree: tree, mapStorage: t.registry.MapStorage}
+}
+
+// singleTXRunner satisfies the merkle.TXRunner interface.
+type singleTXRunner struct {
```

Comment:
Maybe just add a one-line comment on this and `multiTXRunner` describing what they do and why?